jenkins2/Dockerfile

```diff
@@ -0,0 +1,81 @@
+FROM registry.service.opg.digital/opguk/jre-8
+
+### This section is from https://github.com/jenkinsci/docker/blob/master/Dockerfile with some changes ###
+
+ENV JENKINS_HOME /var/jenkins_home
+ENV JENKINS_SLAVE_AGENT_PORT 50000
+
+ARG user=app
+ARG group=app
+ARG uid=1000
+ARG gid=1000
+
+# Jenkins is run with user `jenkins`, uid = 1000
+# If you bind mount a volume from the host or a data container,
+# ensure you use the same uid
+RUN groupmod -g ${gid} -n ${group} app \
+    && usermod -d "$JENKINS_HOME" -u ${uid} -g ${gid} -m -s /bin/bash -l ${user} app
+
+# Jenkins home directory is a volume, so configuration and build history
+# can be persisted and survive image upgrades
+VOLUME /var/jenkins_home
+
+# `/usr/share/jenkins/ref/` contains all reference configuration we want
+# to set on a fresh new installation. Use it to bundle additional plugins
+# or config file with your custom jenkins Docker image.
+RUN mkdir -p /usr/share/jenkins/ref/init.groovy.d
+
+ENV TINI_VERSION 0.14.0
```

Comment:
I was trying to maintain the upstream structure for easy upgrade. Looks like that's not going to fly.